install/gcp/installer/util/utils.py

```diff
@@ -232,37 +232,55 @@ def infer_version(advanced_mode):
     Returns:
         str: Selected Forseti branch.
     """
+    version = None
     return_code, out, err = run_command(
-        ['git', 'rev-parse', '--abbrev-ref', 'HEAD'])
+        ['git', 'symbolic-ref', '-q', '--short', 'HEAD'],
+        number_of_retry=0)
+    # The git command above will return empty if the user is not currently
+    # on a branch.
     if return_code:
         print(err)
         print('Will try to infer the Forseti version instead.')
+
+        return_code, out, err = run_command(
+            ['git', 'describe', '--tags', '--exact-match'])
+        # The git command above will return the tag name if we checked out
+        # a tag, will throw an exception otherwise.
+        if return_code:
+            print(err)
+        else:
+            version = 'tags/{}'.format(out.strip()) if out.strip() else ''
     else:
-        branch = out.strip()
+        version = out.strip()
 
     user_choice = None
-    if not branch or branch == 'master':
-        version = get_forseti_version()
+    if not version or version == 'stable':
+        cur_version = get_forseti_version()
```

Comment:
After looking the change, I think the code's flow can be made even more clean and easier to follow by moving all this logic into the `get_forseti_version()` method like below.  How do you think?

```
def infer_version():
    version = None

    # Get forseti version from current branch.
    return_code, out, err = run_command(
        ['git', 'symbolic-ref', '-q', '--short', 'HEAD'],
        number_of_retry=0)

    if return_code:
        print(err)
        print ('Unable to get forseti version from branch. '
               'Will try to infer the Forseti version instead.')
    else:
        return out.strip()

    # Get forseti version from current tag.
    return_code, out, err = run_command(
        ['git', 'describe', '--tags', '--exact-match'])

    if return_code:
        print ('Unable to get forseti version from tag. '
               'Will try to get the Forseti version from source code.')
    else:
        return 'tags/{}'.format(out.strip())
    
    # Get forseti version from source code.
    version_re = re.compile(constants.VERSIONFILE_REGEX)
    version_file = os.path.join(
        constants.FORSETI_SRC_PATH, '__init__.py')

    with open(version_file, 'rt') as vfile:
        for line in vfile:
            version_match = version_re.match(line)
            if version_match:
                version = version_match.group(1)
                return 'v%s' % version

    # Unable to infer version.
    print('Unable to determine the current Forseti from source_code, please '
          'check https://forsetisecurity.org/faq/ for more '
          'information.')
    sys.exit(1)
``